packages/account/src/providers/transaction-response/transaction-response.ts

```diff
@@ -212,13 +212,34 @@ export class TransactionResponse {
     }
   }
 
+  /**
+   * When a response has been built from an unfetched tx request, it's possible that the tx id
+   * for inputs are not set. This method updates the tx ids for these inputs.
+   *
+   * @param tx - the request to update.
+   */
+  private updateTxIds<TTransactionType = void>(transaction: Transaction<TTransactionType>) {
+    const tx = transaction as Transaction<
+      TransactionType.Script | TransactionType.Create | TransactionType.Blob
+    >;
+
+    tx.inputs.forEach((input) => {
+      if (input.type === InputType.Contract) {
+        // eslint-disable-next-line no-param-reassign
+        input.txID = this.id;
+      }
+    });
+  }
+
   private async getTransaction<TTransactionType = void>(): Promise<{
     tx: Transaction<TTransactionType>;
     bytes: Uint8Array;
   }> {
     if (this.request) {
       const tx = this.request.toTransaction() as Transaction<TTransactionType>;
       this.applyMalleableSubscriptionFields(tx);
+      this.updateTxIds(tx);
+
       return {
         tx,
         bytes: this.request.toTransactionBytes(),
```

Comment:
```suggestion
        bytes: new TransactionCoder().encode(tx),
```

If we're mutating the transaction, surely the bytes from the request are incorrect?